library/std/src/rt.rs

```diff
@@ -77,13 +77,17 @@ unsafe fn init(argc: isize, argv: *const *const u8) {
     unsafe {
         sys::init(argc, argv);
 
-        let main_guard = sys::thread::guard::init();
-        // Next, set up the current Thread with the guard information we just
-        // created. Note that this isn't necessary in general for new threads,
-        // but we just do this to name the main thread and to give it correct
-        // info about the stack bounds.
-        let thread = Thread::new(Some(rtunwrap!(Ok, CString::new("main"))));
-        thread_info::set(main_guard, thread);
+        // FIXME: remove PSP special case
```

Comment:
Why is this special-cased to not run on PSP?

Reply:
Psp was crashing. I suspect it's due to some panic. Anyway I didn't have time to investigate. Plus I don't understand what PSP's threads are. All I know is that they are not concurrent.